### PR TITLE
Support host os independent timezone db

### PR DESCRIPTION
**Pull request by mterron**
This commit adds support for independently maintaining an up-to-date timezone database as per #9

It adds requirements of internet access for the `cronjobber-updatetz` image to download updated files.  The `cronjobber-updatetz` support standard proxy env variables.

### CODE REVIEW

File: deploy/deploy.yaml [hunk 1]
```diff
@@ -36,3 +34,14 @@ spec:
         - name: timezonedb
           mountPath: /usr/share/zoneinfo
           readOnly: true
+      - name: updatetz
```
[the remark this answers is not in the document]
Comment (mterron) [reply deploy/deploy.yaml]: Done, however you are assuming that the host os has a timezone database on `/usr/share/zoneinfo`.

**Comment by hiddeco** [comment]
Changes look good on _digital_ paper, need to give it real spin though, which I will do tonight or tomorrow. Thanks so far :tulip:

File: updatetz/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+	  org.label-schema.schema-version="1.0" \
+	  org.label-schema.name="cronjobber" \
+	  org.label-schema.description="Cronjobber is the Kubernetes cronjob controller patched with time zone support" \
+	  org.label-schema.url="https://github.com/hiddeco/cronjobber" \
+	  org.label-schema.vcs-url="git@github.com:hiddeco/cronjobber" \
+	  org.label-schema.vendor="Hidde Beydals"
+
+COPY updatetz.sh /usr/local/bin
+COPY docker-entrypoint.sh /usr/local/bin
+COPY Dockerfile /etc
```
Comment (hiddeco) [inline updatetz/Dockerfile]: Is there a reason for copying this to `/etc`?

Comment (mterron) [reply updatetz/Dockerfile]: I always do it so anyone that gets the image can see how it was built.

Comment (hiddeco) [reply updatetz/Dockerfile]: Are you aware of `docker history --no-trunc <image>`?

Comment (mterron) [reply updatetz/Dockerfile]: I wasn't, no. That's cool!

**Comment by mterron** [comment]
There are a couple of env variables that might be useful to document:
`TZPV` is the location where the persistent volume for the timezone database is mounted on the `updatetz` container. (`/tmp/zoneinfo` by default)
`REFRESH_PERIOD` is how often the `updatetz` container will look for updates (1 week by default).

This is what a running instance log looks like:
```log
2019-05-08T02:12:44+0000 Local Time Zone database updated to version 2019a on /tmp/zoneinfo
2019-05-08T02:13:15+0000 Local Time Zone database is up to date
2019-05-08T02:13:46+0000 Local Time Zone database is up to date
```

**Comment by hiddeco** [comment]
@mterron I see you made a couple more changes, what is the current state of this PR? Is there anything you need help with?

**Comment by mterron** [comment]
It is as generic as I can make it (all the latest commits removed things
that work in my specific environment but not in a default installation).
Please take it for a spin and see if it works for you.

**Comment by hiddeco** [comment]
@mterron failing CI is not related to the content of my changes, a Google server is having issues.

Please take a look at the changes I made, if they look alright to you I will smash a couple of commits together to keep the `master` history clean and merge it. :sun_with_face: 

I am interested in if the modified `securityContext` config is still failing on OpenShift by the way, so if you have some time to look into this that would be awesome.

File: deploy/deploy-updatetz.yaml [hunk 3]
```diff
@@ -17,6 +17,7 @@ spec:
     spec:
       serviceAccountName: cronjobber
       securityContext:
+        runAsUser: 65534 #nobody
```
Comment (mterron) [inline deploy/deploy-updatetz.yaml]: You should not run stuff as `nobody` in general as that user is reserved for NFS; `daemon` is a more generic user.

File: deploy/deploy-updatetz.yaml [hunk 4]
```diff
@@ -35,16 +36,15 @@ spec:
...
-          #   drop:
-          #   - ALL
-          #   add:
-          #   - FOWNER
+        image: hiddeco/cronjobber-updatetz
+        # NB: the security context configuration below may not work
+        # out of the box on OpenShift
+        securityContext:
+          allowPrivilegeEscalation: false
+          runAsNonRoot: true
```
Comment (mterron) [inline deploy/deploy-updatetz.yaml]: `runAsNonRoot` will never work as far as I understand as it requires a numeric UID  and doing it that way it's probably a bad idea with user namespaces coming soon (tm)

File: deploy/deploy.yaml [hunk 5]
```diff
@@ -17,6 +17,7 @@ spec:
     spec:
       serviceAccountName: cronjobber
       securityContext:
+        runAsUser: 65534 #nobody
```
Comment (mterron) [inline deploy/deploy.yaml]: See https://github.com/hiddeco/cronjobber/pull/10/commits/712602b670271d17b1aa454c47b6cff26d2eeb44#r282682142

File: updatetz/Dockerfile [hunk 6]
```diff
@@ -18,7 +18,7 @@ COPY updatetz.sh /usr/local/bin
 COPY docker-entrypoint.sh /usr/local/bin
 
 WORKDIR /tmp
-USER daemon
+USER nobody
```
Comment (mterron) [inline updatetz/Dockerfile]: See https://github.com/hiddeco/cronjobber/pull/10/commits/712602b670271d17b1aa454c47b6cff26d2eeb44#r282682142

File: Dockerfile [hunk 7]
```diff
@@ -3,8 +3,8 @@ ARG GO_VERSION=1.11
 FROM golang:${GO_VERSION}-alpine AS builder
 
 RUN mkdir /user \
-    && echo 'nobody:x:65534:65534:nobody:/:' > /user/passwd \
-    && echo 'nobody:x:65534:' > /user/group
+    && echo 'daemon:x:2:2:daemon:/:' > /user/passwd \
```
Comment (mterron) [inline Dockerfile]: You don't need this 2 lines. `daemon` is present in the default image

Comment (hiddeco) [reply Dockerfile]: This minimizes the contents to just this user. It is not used by this image in the end but in the `cronjobber` image, see: https://github.com/hiddeco/cronjobber/pull/10/files/61f34620e07d55138c990518e896593cbbb94999#diff-3254677a7917c6c01f55212f86c57fbfL43

Comment (mterron) [reply Dockerfile]: It's not doing anything security wise but doesn't harm either I guess.

**Comment by hiddeco** [comment]
Thanks again for your contribution. I have put out releases of both images, as soon as `0.2.0` has been pushed to the registry, `kubectl apply -f https://raw.githubusercontent.com/hiddeco/cronjobber/master/deploy/deploy-updatetz.yaml` should do the job.

**Pull request merged**